ecosystem/platform/server/app/controllers/welcome_controller.rb

```diff
@@ -6,14 +6,48 @@
 class WelcomeController < ApplicationController
   layout 'it1'
 
-  before_action :ensure_discord!, only: %i[it1]
   before_action :ensure_confirmed!, only: %i[it1]
 
   def index
-    redirect_to it1_path if current_user && current_user.authorizations.where(provider: :discord).exists?
+    @login_dialog = DialogComponent.new
   end
 
   def it1
     redirect_to root_path unless user_signed_in?
+    @steps = [
+      connect_discord_step,
+      node_registration_step,
+      identity_verification_step
+    ].map { |h| OpenStruct.new(**h) }
+    first_incomplete = @steps.index { |step| !step.completed }
+    @steps[first_incomplete + 1..].each { |step| step.disabled = true } if first_incomplete
+  end
+
+  private
+
+  def connect_discord_step
+    {
+      name: :connect_discord,
+      completed: current_user.authorizations.where(provider: :discord).exists?,
+      dialog: DialogComponent.new
+    }
+  end
+
+  def node_registration_step
+    completed = !current_user&.it1_profile&.validator_verified?.nil?
```

Comment:
I think you don't need the `.nil?`- 
```
irb(main):001:0> false.nil?
=> false
irb(main):002:0> true.nil?
=> false
```
as otherwise it's always true, if anything is set?